sonar-db/src/main/java/org/sonar/db/permission/OldPermissionQuery.java

```diff
@@ -0,0 +1,206 @@
+  public static final String OUT = "OUT";
+  public static final Set<String> AVAILABLE_MEMBERSHIP = ImmutableSet.of(ANY, IN, OUT);
+
+  private final String permission;
+  private final String component;
+  private final String template;
+  private final String membership;
+  private final String search;
+
+  // for internal use in MyBatis
+  final String searchSql;
+
+  // max results per page
+  private final int pageSize;
+  // index of selected page. Start with 1.
+  private final int pageIndex;
+  // offset. Starts with 0.
+  private final int pageOffset;
+
+  private OldPermissionQuery(Builder builder) {
+    this.permission = builder.permission;
+    this.component = builder.component;
+    this.template = builder.template;
+    this.membership = builder.membership;
+    this.search = builder.search;
+    this.searchSql = searchToSql(search);
+
+    this.pageSize = builder.pageSize;
+    this.pageIndex = builder.pageIndex;
+    this.pageOffset = (builder.pageIndex - 1) * builder.pageSize;
+  }
+
+  private static String searchToSql(@Nullable String s) {
+    String sql = null;
+    if (s != null) {
+      sql = StringUtils.replace(StringUtils.upperCase(s), "%", "/%");
+      sql = StringUtils.replace(sql, "_", "/_");
+      sql = "%" + sql + "%";
+    }
+    return sql;
+  }
+
+  public String permission() {
+    return permission;
+  }
+
+  /**
+   * Used only for permission template
+   */
+  public String template() {
+    return template;
+  }
+
+  /**
+   * Used on project permission
+   */
+  @CheckForNull
+  public String component() {
+    return component;
+  }
+
+  @CheckForNull
+  public String membership() {
+    return membership;
+  }
+
+  @CheckForNull
+  public String search() {
+    return search;
+  }
+
+  public int pageSize() {
+    return pageSize;
+  }
+
+  public int pageOffset() {
+    return pageOffset;
+  }
+
+  public int pageIndex() {
+    return pageIndex;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private String permission;
+    private String component;
+    private String template;
+    private String membership;
+    private String search;
+
+    private Integer pageIndex = DEFAULT_PAGE_INDEX;
+    private Integer pageSize = DEFAULT_PAGE_SIZE;
+
+    private Builder() {
+    }
+
+    public Builder permission(String permission) {
+      this.permission = permission;
+      return this;
+    }
+
+    public Builder template(String template) {
+      this.template = template;
+      return this;
+    }
+
+    public Builder component(@Nullable String component) {
+      this.component = component;
+      return this;
+    }
+
+    public Builder membership(@Nullable String membership) {
+      this.membership = membership;
+      return this;
+    }
+
+    public Builder search(@Nullable String s) {
+      this.search = StringUtils.defaultIfBlank(s, null);
+      return this;
+    }
+
+    public Builder pageSize(@Nullable Integer i) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)